tpblite/models/utils.py

```diff
@@ -1,57 +1,80 @@
+from typing import Tuple, Type
 import random
 from urllib.request import Request, urlopen
 import urllib.error
 from purl import URL as pURL
 
 
-class QueryParser(object):
-    '''
-    Query object capable of getting html response given 
-    a search query and other parameters.
-    '''
-    def __init__(self, query, base_url, page, order, category):
+class QueryParser:
+    """Query object capable of getting html response given a search query and other
+    parameters.
+    """
+
+    # PirateBay URL to use for queries
+    base_url: str
+
+    # Compiled search string used to query the PirateBay URL
+    url: str
+
+    def __init__(self, base_url: str, segments: Tuple[str, ...]):
         self.base_url = base_url
-        segments = ('search', query, str(page), str(order), str(category))
         self.url = URL(base_url, segments)
         try:
             self.html_source = self._sendRequest()
         except urllib.error.URLError:
-            raise ConnectionError('Could not establish connection wtih {}'.format(self.base_url))
-     
+            raise ConnectionError(
+                "Could not establish connection wtih {}".format(self.base_url)
+            )
+
+    @classmethod
+    def from_search(
+        cls, query: str, base_url: str, page: int, order: int, category: int
+    ):
+        segments = ("search", query, str(page), str(order), str(category))
+        return cls(base_url, segments)
+
+    @classmethod
+    def from_browse(cls, base_url: str, category: int, page: int, order: int):
+        print("browsing")
+        segments = ("browse", str(category), str(page), str(order), "0")
```

Comment:
the "0" at the end is not necessary, the /browse URL will work without specifying the category at the end (as you already specify the category right after /browse/)

Reply:
ah? when I "browse" directly on the website, the 0 is automatically added to the URL everytime. The URL works without the 0 on the website, but if we don't add it in the segments, will it result in an additional web request to follow the URL? I have no idea, I'll trust you on this

Reply:
> when I "browse" directly on the website, the 0 is automatically added to the URL everytime.

Hm, you're right it does. It wouldn't make an additional request i don't think. Since it's consistent with the manual web request let's keep the "0".

Reply:
Comment added in the code to explain.